pkg/capture/metrics.go

```diff
@@ -88,3 +88,14 @@ func init() {
 		promRotationDuration,
 	)
 }
+
+// ResetCounters allows to externally reset all Prometheus counters (e.g. for testing purposes
+// or in order to manually reset all of them)
+func ResetCounters() {
```

Comment:
Not that I'm aware of.

What's important to note on the function is that it _MUST NOT_ be used outside the testing code, so I would even flip it, make it a private method inside the testing package. Accept a list of collectors as argument and run reset on those.

Monotonically increasing counters should not be reset.

So maybe something like
```golang
func reset(collectors ...prometheus.Collector) {}
```
would work.

Reply:
> Not that I'm aware of.
> 
> What's important to note on the function is that it _MUST NOT_ be used outside the testing code, so I would even flip it, make it a private method inside the testing package. Accept a list of collectors as argument and run reset on those.
> 
> Monotonically increasing counters should not be reset.
> 
> So maybe something like
> 
> ```go
> func reset(collectors ...prometheus.Collector) {}
> ```
> 
> would work.

I agree, if it only were that simple. As I hinted I need to call this _from the E2E tests_, which won't have access to any private function / method / object (so no matter what I do, _something_ has to be publicly made available.

I'll have to think about this, maybe allowing to specify a custom registry isn't such a bad idea, that way at least it's explicit and I can create that from the E2E tests (where I programmatically setup a custom goProbe anyway).

Reply:
@els0r To clarify: You cannot call stuff limited to `_test` in one package from another, this has been discussed (and rejected) in the past, e.g.:

- https://github.com/golang/go/issues/39565
- https://github.com/golang/go/issues/31135

But I found an acceptable solution: Since Go 1.22 you can do `testing.Testing() bool`, which will return `true` if we are running from tests. I've added this as a guard in the exposed `ResetCounters()` (in addition to a comment) function which will panic if called from outside the testing facilities.